cmake/YarpVersion.cmake

```diff
@@ -11,6 +11,10 @@ include(GitInfo)
 
 set(YARP_VERSION_ABI "1")
 
+set(YARP_PROTOCOL_VERSION_MAJOR "1")
+set(YARP_PROTOCOL_VERSION_MINOR "2")
+set(YARP_PROTOCOL_VERSION_PATCH "4")
```

Comment:
All three of them were already present, I just moved them. I have no opinion about using 2 or 3 number, I just have not found any reason to remove it. Instead, I think that when a small change is done and it does not break the compatibility (like an optimization or similar), it can be done without enforcing the user to update the code.

The point here however is not simply add or remove messages, consider also case in which the message is the same, but the binary representation change. Like it is gonna happen to envelope soon.

Reply:
> All three of them were already present, I just moved them.

Yes, I know, and I was probably the one who suggested that in the first place, to maintain compatibility, sorry about that...
But since we are breaking things, I suggest we handle this properly...
Usually you have 3 number to represent versions, the third is the "implementation" level (private), but on a *protocol* you just have a "public" level, therefore I believe that "MAJOR.MINOR" is enough (also it is what actual protocol use, see HTTP, for example)

> The point here however is not simply add or remove messages, consider also case in which the message is the same, but the binary representation change. Like it is gonna happen to envelope soon.

You are making the example of a *very* breaking change...
If the binary representation changes, that's one of the cases when the `MAJOR` number should be increased...

---

[OT] In the case of the envelope, are you sure you have  I think the way to go would be to add a new message that specifies that you are sending a message with a binary envelope, that means a `MINOR` change. We should discuss this...